src/crystal/hasher.cr

```diff
@@ -118,7 +118,7 @@ struct Crystal::Hasher
   end
 
   def self.reduce_num(value : Int)
-    value.remainder(HASH_MODULUS).to_i64.unsafe_as(UInt64)
+    value.remainder(HASH_MODULUS).to_u64!
   end
```

Comment:
**nitpick**: this is slightly different: in the very odd case that `value` is an Int128 then the conversion might overflow and this used to raise, as will the above overload with Int::Unsigned :grin:

Reply:
The result of `remainder(HASH_MODULUS)` is in the range `0...HASH_MODULUS`, and thus guaranteed to fit into `UInt64` (`HASH_MODULUS` is `Int64`).

Reply:
That would be `#%`, the range for `#remainder` is `-HASH_MODULUS + 1..HASH_MODULUS - 1`

Reply:
Oh right. Doesn't change the fact that the result fits in 64-bits and no overflow when casting to `UInt64`.